### PR TITLE
Upgrading to latest rustc

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: scripts/rust-linter.sh [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    -A clippy::question-mark \
+    -A clippy::result-large-err \
+    -A clippy::single-char-pattern \
+    -A clippy::type-complexity \
+    -A clippy::uninlined-format-args \
+    -A clippy::unnecessary-cast \
+    -A clippy::unnecessary-lazy-evaluations \
+    -A clippy::unnecessary-to-owned \
+    -A clippy::unwrap-or-else-default \
+    -A clippy::useless-conversion
```
Comment (tiagolobocastro) [inline scripts/rust-linter.sh:26]: Are all these allows necessary now? :(

**Comment by dsavitskiy** [comment]
bors try

File: scripts/rust-linter.sh (hunk 1 above)
Comment (dsavitskiy) [reply scripts/rust-linter.sh:26]: These are new lints introduced by the latest clippy. Some of them look quite useful like "await-holding-lock". I'm going to create a separate ticket to fix them.

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Review by tiagolobocastro — APPROVED** [review]

**Review by mtzaurus — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**